tools/python/stylesheet/color_matrix_builder.py

```diff
@@ -0,0 +1,244 @@
+        tup_color_from_img = self.imgs[path_index].img.getpixel(xy)
+        assert tup_color == tup_color_from_img, "Colors in the image and the bin file are diffrent"
+
+
+    # CORE
+    def foreach_path(self, fn):
+        map(fn, xrange(len(self.paths)))
+
+
+    def color_key_generator(self, sub_drules, index_list):
+        return [
+            (index_list + "/" + color_key, color_key, getattr(sub_drules, color_key))
+            for color_key in color_fields(sub_drules)
+        ]
+
+
+    def recursive_iterator(self, sub_drules, index_list, index, worker_fn):
+        if has_color_fields(sub_drules):
+            worker_fn(sub_drules, index_list)
+
+        for key, element in list_message_fields(sub_drules):
+            self.recursive_iterator(element, index_list + key, index, worker_fn)
+
+
+    def normalize_all_keys(self):
+        self.foreach_path(lambda x: self.normalize_keys_at_index(x))
+        print("Hello")
+
+
+    def normalize_keys_at_index(self, path_index):
+        priority_key_transformation_dict = {}  # new_key: old_key
+
+        keys_to_delete = []
+        new_keys_dict = {}  # str -> color
+        new_priorities_dict = defaultdict(dict)  # {rest_of_key: {priority: color}}
+
+        for old_key, val in self.all_elements[path_index].iteritems():
+            key = old_key
+
+            priority_parts = PRIORITY_RE.findall(key)
+            if priority_parts:
+                rest_of_key, int_priority = self.make_priority_key_parts(priority_parts)
+                new_priorities_dict[rest_of_key][int_priority] = val
+                keys_to_delete.append(key)
+                priority_key_transformation_dict[self.priority_key(rest_of_key, int_priority)] = old_key
+            else:
+                self.key_correspondences[path_index][key] = key
+
+        for rest_of_key, priorities in new_priorities_dict.iteritems():
+            for i, (priority, color) in enumerate(sorted(priorities.iteritems())):
+                final_key = self.priority_key(rest_of_key, i)
+                new_keys_dict[final_key] = color
+                self.key_correspondences[path_index][
+                    priority_key_transformation_dict[self.priority_key(rest_of_key, priority)]] = final_key
+
+        self.update_all_elements(keys_to_delete, new_keys_dict, path_index)
+
+
+    def make_priority_key_parts(self, priority_parts):
+        priority_parts = priority_parts[0]
+        rest_of_key = "{}/{}".format(priority_parts[0], priority_parts[2])
+        int_priority = int(priority_parts[1])
+        return rest_of_key, int_priority
+
+
+    def priority_key(self, rest_of_key, i):
+        return "{}~{}".format(rest_of_key, i)
+
+
+    def update_all_elements(self, keys_to_delete, new_keys_dict, path_index):
+        for key in keys_to_delete:
+            self.all_elements.pop(key, None)
+        self.all_elements[path_index].update(new_keys_dict)
+
+
+def main():
+    if len(sys.argv) < 2:
```

Comment:
No. Argparse would make more sense if you have any work with the cmd. Like printing help message. But for this case it is not crucial, I agree.